### PR TITLE
GPIO api updated

### PR DESCRIPTION
**Pull request by mudassar-ublox**
Notes:
* Pull requests will not be accepted until the submitter has agreed to the [contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md).
* This is just a template, so feel free to use/remove the unnecessary things

## Description
A few sentences describing the overall goals of the pull request's commits.

## Status
**READY/IN DEVELOPMENT/HOLD**

## Migrations
If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

YES | NO

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos
- [ ] Tests
- [ ] Documentation

## Deploy notes
Notes regarding the deployment of this PR. These should note any
required changes in the build environment, tools, compilers, etc.

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by mudassar-ublox** [comment]
GCC_ARM Build,
http://jenkins.pk-lhr.u-blox.net:8080/job/mbed/job/KM/job/Pull%20request%20build/29/console

**Comment by mudassar-ublox** [comment]
Build Result: http://jenkins.pk-lhr.u-blox.net:8080/job/mbed/job/KM/job/Pull%20request%20build/33/console

Build is successful.

File: targets/TARGET_ublox/TARGET_R5030/gpio_api.c [hunk 1]
```diff
@@ -18,64 +18,183 @@
...
+    	   return GPIO_4_CHANNEL;
+
+       case LED5:
+    	   return GPIO_5_CHANNEL;
+
+       case LED6:
+    	   return GPIO_6_CHANNEL;
+
+       default:
+    	   return ERROR;
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/gpio_api.c]: There are 16 GPIOs KM

File: targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h [hunk 2]
```diff
@@ -54,18 +54,25 @@ typedef enum {
...
-    LED3 = p4,
-    LED4 = p5,
+
+    LED0 = p0,
+    LED1 = p1,
+    LED2 = p2,
+    LED3 = p3,
+    LED4 = p4,
+    LED5 = p5,
+    LED6 = p6,
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h:75]: I am unsure that there will be 7 LED with KM, we got 0-6 with LED on FPGA just for testing. @RobMeades  can you please confirm ?

File: targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h [hunk 3]
```diff
@@ -100,10 +107,11 @@ typedef enum {
 typedef enum {
     PullNone = 0,
     PullDown = 1,
-    PullUp = 0xff,  // TODO: current HI2110 chip does not have pull-up 
+    PullUp = 2,  // TODO: current HI2110 chip does not have pull-up
     PullDefault = PullDown // TODO: mbed requires PullUp as default but HI2110 doesn't have it
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h]: Can you please update comment as PullDown is default

Comment (mudassar-ublox) [reply targets/TARGET_ublox/TARGET_R5030/TARGET_R5XXX/PinNames.h]: ok

File: targets/TARGET_ublox/TARGET_R5030/gpio_api.c [hunk 4]
```diff
@@ -18,64 +18,183 @@
...
+      case PullDown:
+    	  obj->reg_base->pio_pler_0 = obj->mask; // Set pull down
+    	  break;
+
+      case PullUp:
+    	  obj->reg_base->pio_pher_0 = obj->mask; // Set pull up
+    	  break;
+
+      default:
+    	  break;
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/gpio_api.c:84]: Please add a meaning full trace here

File: targets/TARGET_ublox/TARGET_R5030/gpio_api.c (hunk 1 above)
Comment (fahimalavi) [reply targets/TARGET_ublox/TARGET_R5030/gpio_api.c]: You are returning GPIO_0_CHANNEL and ERROR, both have value 0. May be in default case you want to add print and assign some other value than GPIO_0_CHANNEL

File: targets/TARGET_ublox/TARGET_R5030/gpio_api.c [hunk 5]
```diff
@@ -18,64 +18,183 @@
...
-    obj->pin = pin;
-    obj->mask = (1ul << pin);
+	obj->pin = pin;
 
-    // Update it when gpio driver is ready
-    //obj->reg_set = &GPIO_OUT_BITSET;
-    //obj->reg_clr = &GPIO_OUT_BITCLR;
-    //obj->reg_val = &GPIO_VALUE;
-   // obj->reg_dir = &GPIO_DIR;
+	pioChannel = gpio_channel_select(obj->pin);
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/gpio_api.c]: What if this method returns value ERROR ? Please add a check

File: targets/TARGET_ublox/TARGET_R5030/gpio_api.c [hunk 6]
```diff
@@ -18,64 +18,183 @@
...
-         * not normally trapped here */
+    switch (direction)
+    {
+      case PIN_INPUT:
+    	  obj->reg_base->pio_iner_0 = obj->mask; // Enable input
+    	  break;
+
+      case PIN_OUTPUT:
+    	  obj->reg_base->pio_oer_0 = obj->mask; // Enable output
+    	  break;
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/gpio_api.c:104]: Please add a default case

**Review by fahimalavi — APPROVED** [review]

**Pull request merged**